civis/polling.py

```diff
@@ -104,6 +117,64 @@ def __repr__(self):
                                             self._civis_state)
             return out
 
+    def _subscribe(self):
+        pubnub = None
+        if self.poller_args:
+            class JobCompleteListener(SubscribeCallback):
+                def __init__(self, job_id, callback_function):
+                    self.job_id = job_id
+                    self.callback_function = callback_function
+
+                def message(self, pubnub, message):
+                    try:
+                        result = message.message
+                        if result['object']['id'] == self.job_id \
```

Comment:
Does the pubnub message include `run_id`? If so, it would be nice to check against both job and run ids to be as precise as possible.